File: Go/14/14_test.go
package longest_common_prefix

import (
	"leetcode/pkg/assert"
	"testing"
)

func TestLongestCommonPrefix(t *testing.T) {
	cases := []struct {
		name string
		strs []string
		want string
	}{
		{
			name: "CommonPrefixInMultipleWords",
			strs: []string{"flower", "flow", "flight"},
			want: "fl",
		},
		{
			name: "NoCommonPrefix",
			strs: []string{"dog", "racecar", "car"},
			want: "",
		},
	}

	for _, tt := range cases {
		t.Run(tt.name, func(t *testing.T) {
			actual := longestCommonPrefix(tt.strs)
			assert.Equal(t, actual, tt.want)
		})
	}
}

File: JavaScript/tests/14.test.js
import { longestCommonPrefix } from "../src/14.js";
import { assertEquals } from "jsr:@std/assert";

const cases = [
	{ strs: ["flower", "flow", "flight"], want: "fl" },
	{ strs: ["dog", "racecar", "car"], want: "" },
];

Deno.test("14. Longest Common Prefix", () => {
	for (let i = 0; i < cases.length; i++) {
		const result = longestCommonPrefix(cases[i].strs);
		assertEquals(result, cases[i].want);
	}
});
